### ISSUE TITLE
creation of series folder may contain stale info

### ISSUE THREAD

**Issue by thezoggy** [Sonarr/Sonarr#6758]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

When sonarr creates series folder it uses the data when the show was added to sonarr rather than the actual current data. 
Even if you have it set to not create empty folders in advanced, thus it does not create the actual series folder until the first time something is imported.

This can create some pain if the series name or year has chanced since originally adding the series.

### Describe the solution you'd like

If the series folder does not exist, use updated data when creating.

### Describe alternatives you've considered

Many people mask this issue with using imdb/tvdbid in their series folder, however not all apps support this at this time.

### Anything else?

this has been a pretty common complaint over the years in radarr land where a lot of stuff is announced well in advanced, it still however does impact tv series from time to time.

**Comment by thezoggy** [comment]
and just so its crystal clear, this is purely to deal with the scenario where the user has not created the series folder.. as instructed not to by sonarr and no files have been imported. 

i completely agree with not making a new series folder/renaming anything existing. thus if someone was to create a folder in advanced or the stuff getting modified after the fact.. that would be a health check scenario. this is not that.

**Comment by bakerboy448** [comment]
TL;DR do the equivalent of a same-root-folder move and outcome is recalculate the folder name if the folder does not yet exist and only for folder creation....do not rename existing folders

**Comment by markus101** [comment]
The path is set for the series when it's added, regardless of whether the series folder is created.

No plans to post-pone defining the path until the series folder is created.
